src/mappings/handlers/multisigEventHandler.ts

```diff
@@ -0,0 +1,157 @@
+import { SubstrateEvent } from "@subql/types";
+import { u8aToHex } from "@polkadot/util";
+import { decodeAddress } from "../../utils";
+import { CreateCallVisitorBuilder, CreateCallWalk, VisitedCall } from "subquery-call-visitor";
+import { EventStatus, MultisigEvent, MultisigOperation, OperationStatus } from "../../types";
+import {
+  generateEventId,
+  generateOperationId,
+  getBlockCreated,
+  getDataFromCall,
+  getDataFromEvent,
+  getIndexCreated,
+  timestamp,
+} from "../../utils/operations";
+import { AccountId, DispatchResult, Timepoint } from "@polkadot/types/interfaces";
+import { AnyTuple, CallBase } from "@polkadot/types/types";
+
+const callWalk = CreateCallWalk();
+
+export async function handleMultisigCallEvent(event: SubstrateEvent) {
```

Comment:
Imho using single method to process initialization, approval, rejection and execution makes this method much harder to read because you cannot easily understand which one of them a particular piece code correponds to.
For example, updating blockCreated and indexCreated in the cal visitor does not make much sense for everything - this data should only be filled for initialization event

Thus said, I suggest to define 4 different handlers explicitly and share some common-logic between them, though I dont expect that there will be many things to share